server/src/main/kotlin/suwayomi/tachidesk/manga/controller/UpdateController.kt

```diff
@@ -93,11 +93,20 @@ object UpdateController {
         if (clear) {
             updater.reset()
         }
-        categories
+
+        val categoriesToExclude = categories.filter { !it.includeInUpdate }
+        val unfilteredMangasToUpdate = categories
+            .filter { it.includeInUpdate }
             .flatMap { CategoryManga.getCategoryMangaList(it.id) }
             .distinctBy { it.id }
+        val mangasToCategoriesMap = CategoryManga.getMangasCategories(unfilteredMangasToUpdate.map { it.id })
+        val mangasToUpdate = unfilteredMangasToUpdate.filter {
+            it.updateStrategy == UpdateStrategy.ALWAYS_UPDATE &&
+                !categoriesToExclude.any { category -> mangasToCategoriesMap[it.id]!!.contains(category) }
```

Comment:
No need to enforce not null
```suggestion
                !categoriesToExclude.any { category -> mangasToCategoriesMap[it.id]?.contains(category) == true }
```